### PR TITLE
Fix 4399 Export map doesn't work anymore

### PR DESCRIPTION
**Pull request by kasongoyo**
## Description
Fix map export functionality. This https://github.com/geosolutions-it/MapStore2/pull/4343/ introduced this bug, it added the requirement of passing background layers in saveMapConfiguration so it was necessary for all parts of code to be updated with that changes.

## Issues
 - #4399 
 - ...

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
see #4399 

**What is the new behavior?**
Map export works

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes, and I documented them in migration notes
 - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/epics/__tests__/mapexport-test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const expect = require('expect');
```
Comment (MV88) [inline web/client/epics/__tests__/mapexport-test.js]: please use es6 import syntax

File: web/client/epics/mapexport.js [hunk 2]
```diff
@@ -13,11 +13,12 @@ const { setControlProperty } = require('../actions/controls');
 
 const { mapSelector } = require('../selectors/map');
 const { layersSelector, groupsSelector } = require('../selectors/layers');
+const { backgroundListSelector } = require('../selectors/backgroundselector');
 const { mapOptionsToSaveSelector } = require('../selectors/mapsave');
 const textSearchConfigSelector = state => state.searchconfig && state.searchconfig.textSearchConfig;
 
 const PersistMap = {
-    mapstore2: (state) => JSON.stringify(MapUtils.saveMapConfiguration(mapSelector(state), layersSelector(state), groupsSelector(state), textSearchConfigSelector(state), mapOptionsToSaveSelector(state)))
+    mapstore2: (state) => JSON.stringify(MapUtils.saveMapConfiguration(mapSelector(state), layersSelector(state), groupsSelector(state), backgroundListSelector(state), textSearchConfigSelector(state), mapOptionsToSaveSelector(state)))
```
Comment (MV88) [inline web/client/epics/mapexport.js]: please add a catch method where you dispatch a localized message, in a notification action, that will tell user something as gone wrong and prevent the app to stuck or throw an error

Comment (MV88) [inline web/client/epics/mapexport.js]: why you added backgroundListSelector here?

File: web/client/epics/__tests__/mapexport-test.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+const { testEpic } = require('./epicTestUtils');
+const { exportMapContext: exportMapContextEpic } = require('../mapexport');
+const { exportMap } = require('../../actions/mapexport');
+const { SET_CONTROL_PROPERTY } = require('../../actions/controls');
+
+describe('Map Export Epics', () => {
+    it('export map', (done) => {
```
Comment (MV88) [inline web/client/epics/__tests__/mapexport-test.js]: when adding the catch method add a test for that case also

File: web/client/epics/__tests__/mapexport-test.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        const epicResult = actions => {
+            expect(actions.length).toBe(1);
+            const action = actions[0];
+            expect(action.type).toBe(SET_CONTROL_PROPERTY);
```
Comment (MV88) [inline web/client/epics/__tests__/mapexport-test.js]: please, check also for other action property not only the type

File: web/client/epics/mapexport.js (hunk 2 above)
Comment (kasongoyo) [reply web/client/epics/mapexport.js]: It was introduced by https://github.com/geosolutions-it/MapStore2/pull/4343/files#

**Pull request merged**